### DISCUSSION TITLE
My model from "river.reco" does not find 'user' and 'item' arguments in the MovieLens dataset

### DISCUSSION THREAD

**Discussion by TxusLopez** [online-ml/river#1499] [Q&A] [answered]
Hi,

I am trying to follow the example of "Matrix factorization for recommender systems", but I get this error in the Part 1 for the Baseline model:

> TypeError: predict_one() missing 2 required positional arguments: 'user' and 'item'

Here the code:

```
import json
import itertools
import pandas as pd

from river import datasets
from river import metrics
from river import evaluate
from river import dummy
from river import stats
from river import preprocessing
from river import optim
from river import reco
from river import compose
from river import facto
from sklearn.preprocessing import StandardScaler


#MovieLens
for x, y in datasets.MovieLens100K():
    print(f'x = {json.dumps(x, indent=4)}')
    print(f'y = {y}')
    break

X_y = datasets.MovieLens100K(unpack_user_and_item=False)

############################ Dummy #########################################

dmmy_metrics = metrics.MAE() + metrics.RMSE()
#MovieLens
dummy_model = dummy.StatisticRegressor(stats.Mean())
dummy_steps = evaluate.iter_progressive_val_score(dataset=X_y, model=dummy_model, metric=dmmy_metrics, step=1)

for dstep in itertools.islice(dummy_steps, 100, 105):
   print(dstep)

############################ Baseline #########################################

baseline_metrics = metrics.MAE() + metrics.RMSE()
baseline_params = {
    'optimizer': optim.SGD(0.025),
    'l2': 0.,
    'initializer': optim.initializers.Zeros()
}

#MovieLens
baseline_model = preprocessing.PredClipper(
    regressor=reco.Baseline(**baseline_params),
    y_min=1,
    y_max=5
)

baseline_steps = evaluate.iter_progressive_val_score(dataset=X_y, model=baseline_model, metric=baseline_metrics, step=1)

for bstep in itertools.islice(baseline_steps, 100, 105):
   print(bstep)
```

Thanks in advance!

**Comment by gbolmier** [comment]
Hey @TxusLopez, according to the [docs](https://riverml.xyz/latest/api/reco/Baseline/#methods) `river.reco` models `learn_one` and `predict_one` methods take a `user` and an `item` inputs.

Try loading your dataset this way in your code before calling `evaluate.iter_progressive_val_score` and it should work:

```py
X_y = datasets.MovieLens100K(unpack_user_and_item=True)
```

**Comment by TxusLopez** [comment, marked as answer]
Thanks @gbolmier for your response. I get the same error with that. But I was trying to do the same without using the `evaluate.iter_progressive_val_score` by doing the process manually with a simple iteration and detailing the params `x,y,user,item`:

```
#Training
for index, row in X_train.iterrows():
    user = row['user']
    item = row['item']
    feats = row.to_dict()
    targ = y_train[index]

    baseline_model = baseline_model.learn_one(x=feats, y=targ, user=user, item=item)
```
, being `X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=42)` and 

```
X = df_movielens.drop('rating', axis=1)  
y = df_movielens['rating']       
```       
, where `df_movielens` is just a DataFrame of the MovieLens dataset. And then I got a code without that error.

**Reply by gbolmier** [reply]
Works on my machine though 😉

<img width="1049" alt="Screenshot 2024-02-02 at 3 07 00 PM" src="https://github.com/online-ml/river/assets/25319692/8e6996a1-1aa9-401b-97dc-d4d57d1f74a6">
